Add RunnerSelectionTests for 'multiple project' packages

ChrisMaddock opened the pull request:
N.B. To be reviewed after #597, branched off that one

-----
Next step in #514. As before, tests reflecting current behaviour, rather than correct behaviour!

jnm2 commented on src/NUnitEngine/nunit.engine.tests/Services/DefaultTestRunnerFactoryTests.cs:
```diff
@@ -55,48 +55,22 @@ public void ServiceIsStarted()
         }
 
+        //Note: These tests currently overlap with RunnerSelectionTests, and are being removed as they are
+        //re-implemented in that test class
```
Least important thing imaginable- for consistency, you could add a space between `//` and the following text if you want to. :D

ChrisMaddock commented:
As https://github.com/nunit/nunit-console/pull/597 - this needs a GitHub review to merge - but the changes since the last review are trivial.

(Thanks Joseph for reviewing!)
